src/grammar_analyzer.py

```diff
@@ -0,0 +1,34 @@
+# pylint: disable=missing-docstring
+import language_tool_python
+import re
+from typing import List, Tuple
+
+# Mention the language keyword
+
+
+def grammar_analyzer(text: str) -> List[Tuple[int, int]]:
+    '''A tool to check grammar error and grade it in the reflection'''
+    tool = language_tool_python.LanguageTool('en-US')
+
+    # Variable represent number of grammar errors in the text
+    err_num = 0
+
+    # count for grammar errors
+    for line in text:
+        matches = tool.check(line)
+        err_num = err_num + len(matches)
+
+    # Store all alphanumeric characters in the reflection in a list
+    words = re.sub('[^0-9a-zA-Z]+', ' ', str(text)).lower().split()
```

Comment:
I'm not great at regex, although, is this line tokenizing the text?

Reply:
This line is expected to replace all non alphanumeric character in the text with white space and then tokenize that text, store under the list call words.

Reply:
Okay, thanks! That's what I thought! If that's the case, the tokens are already processed and stored in the data frame when markdown documents are being imported. See if you can just reuse them instead of retokenizing from the text.